docs/rst/enhancements/keys/topic-keys.rst

```diff
@@ -5,49 +5,80 @@ Topic Keys
+Instead of having multiple topics, one per robot, topic keys allow to have a single topic where each robot sends its status with a unique key (the robot id) that identifies the robot.
+That is the power of topic instances.
+
+.. list-table::
+   :width: 100%
+   :class: borderless
+
+   * - .. image:: ../../figures/enhancements/keys/topic_instances_example.svg
+          :width: 100%
+
+     - .. image:: ../../figures/enhancements/keys/topics_example.svg
+          :width: 100%
+
+In addition, the use of topic instances report numerous advantages, including:
+
+* Efficient middleware infrastructure, data distribution and resource utilization as the number of entities are dramatically reduced.
+* A better history management, meaning that cached updates are managed in a more performant way for determined Quality of Service parameters.
+
+These benefits are further detailed in the :ref:`benefits_of_topic_keys` section.
 
 Background
 ^^^^^^^^^^
 *Vulcanexus* uses :ref:`Fast DDS as middleware <vulcanexus_middleware>`.
-One of the core concepts involved in the communication architecture are topics,
-which serve as the bridge that enable data exchange among nodes within a particular system.
+One of the core concepts involved in the *Vulcanexus* communication architecture is topics,
+which serve as the bridge that enables data exchange among nodes within a particular system.
 
-Topics are named communication channels that allow nodes (software modules) in a robotic system
+Topics are named communication channels that allow nodes (software modules) in a distributed system (such as a robotic application)
 to exchange messages with each other. These messages can contain various types of data, such as sensor readings,
 control commands, or status updates, enabling different parts of the robot to collaborate and share information.
 
-The following image depicts a simple scenario where two sensors publish data in different topics, and two nodes,
-the first ``Logging`` node subscribes to the Sensor 2 output topic, and the ``Controller`` node that subscribes
+The following image depicts a simple scenario where two sensors publish data in different topics, and two remote nodes subscribe to it.
+The first ``Logging`` node only subscribes to the Sensor 2 output topic, while the ``Controller`` node subscribes
 to both topics.
 
 .. image:: ../../figures/enhancements/keys/topics.gif
 
 When any new data is available in the sensors, it is transmitted through the corresponding topic and all the subscribers
 attached to that topic receive the message.
 
+In this situation, each of the sensors has a dedicated topic for transmitting its data. This suggests excessive resource usage because the middleware
+has to create and manage additional entities (topics) and can potentially lead to a discovery overhead as the number of sensors scale up.
```

Comment:
```suggestion
has to create and manage additional entities (topics, publishers, and subscriptions) and can potentially lead to a discovery overhead as the number of sensors scale up.
```

Reply:
NIT(I referred to as  `publications` instead)

Reply:
The thing is that ROS 2 lingo is Publisher and Subscription 🤷🏼